cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/StandaloneMetricDatasetFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metrics.store;
+
+import co.cask.cdap.api.dataset.table.TableProperties;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.dataset2.lib.table.MetricsTable;
+import co.cask.cdap.data2.dataset2.lib.table.hbase.HBaseTableAdmin;
+import co.cask.cdap.data2.dataset2.lib.timeseries.FactTable;
+import com.google.inject.Inject;
+
+/**
+ * Standalone metrics dataset factory
+ */
+public class StandaloneMetricDatasetFactory extends DefaultMetricDatasetFactory {
+
+  @Inject
+  public StandaloneMetricDatasetFactory(CConfiguration cConf, DatasetFramework dsFramework) {
+    super(cConf, dsFramework);
+  }
+
+  @Override
+  public FactTable getOrCreateFactTable(int resolution) {
+    String tableName = cConf.get(Constants.Metrics.METRICS_TABLE_PREFIX,
+                                 Constants.Metrics.DEFAULT_METRIC_TABLE_PREFIX) + ".ts." + resolution;
+    int ttl = cConf.getInt(Constants.Metrics.RETENTION_SECONDS + "." + resolution + ".seconds", -1);
+
```

Comment:
Looks like this is copying more code than needed. In DefaultMetricDatasetFactory.getOrCreateFactTable(), if you remove the v2TableName and move that to getOrCreateResolutionMetricsTable(), then all this class needs to do is:
```
@Override
public MetricsTable getOrCreateResolutionMetricsTable(String tableName,
                                                      TableProperties.Builder props) {
  return getOrCreateMetricsTable(tabeName, props.build());
}  
```